cmd/web3/main.go

```diff
@@ -229,6 +229,68 @@ func main() {
 				IncreaseGas(ctx, privateKey, network, c.String("tx"), c.String("amount"))
 			},
 		},
+		{
+			Name:  "replace",
+			Usage: "Replace transaction. If a transaction is still pending, you can attempt to replace it.",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:        "private-key, pk",
+					Usage:       "The private key",
+					EnvVar:      pkVarName,
+					Destination: &privateKey,
+					Required:    true},
+				cli.StringFlag{
+					Name:     "nonce",
+					Usage:    "The nonce to replace.",
+					Required: true},
+				cli.StringFlag{
+					Name:     "to",
+					Usage:    "to address",
+					Required: true,
+				},
+				cli.StringFlag{
+					Name:     "amountd", // adding a d for backwards compatibility. If d, then it's decimal, otherwise, the old stuff.
```

Comment:
Is this necessary? Upgrading from int to allow decimal is already backwards compatible.

Reply:
It uses base by default (wei) doesn't it? So it's not compatible.

Reply:
Oh, if the difference is actual units, not just format, then a different flag make sense to me, but should it be more descriptive? We've been calling this the 'base' unit. Would `wei`/`base` flags be more readable?
```
-amount 10
-amountd 0.001
-wei 10
-base 0.001
```

Reply:
Can't really use wei since that doesn't apply to all tokens. I thought base was the synonym for wei, or no?  Base is the decimal version? 

If I were to start this project from scratch, I'd probably just use the decimal version only and keep "amount" as the name for that, but alas, we're here. amountd seemed appropriate for the decimal version.

Reply:
"Base" was originally meant to be a generic way of saying Eth/Go. I'm not sure what assumptions there are in regard to tokens. However, IMHO using 'decimal' to imply a certain unit is very confusing. Decimal format can be accepted for base units, gwei, tokens, etc. Only wei or raw token values must be integers.

Reply:
> If I were to start this project from scratch, I'd probably just use the decimal version only and keep "amount" as the name for that, but alas, we're here

Since we are still pre-1.0 and can just bump the minor version for breaking changes, my vote would be to nuke/change whatever is necessary to keep everything consistent.

Reply:
We also have https://godoc.org/github.com/gochain/web3#ParseAmount, which parses the units from the string. That could be expanded with decimal support.

Reply:
Using 'decimal' to imply the real unit of the token makes the most sense to me. Right now we're implying that everything is the intified (new word ©️ ™️ ) value (decimal value multiplied by decimals) which is very cumbersome for a human to work with. We use gwei only in terms of gas prices and that's because that's what everyone uses so that should probably stay. 

My worry with a change like making decimal format the default using the same flags is that if we change it and someone upgrades then sends 1000000000000000000 ETH (1 ETH in decimal format), he/she will be sending 1e18 ETH. That's a pretty scary breaking change.

Reply:
> My worry with a change like making decimal format the default using the same flags is that if we change it and someone upgrades then sends 1000000000000000000 ETH (1 ETH in decimal format), he/she will be sending 1e18 ETH. That's a pretty scary breaking change.

I agree. To be clear I am absolutely not advocating that the units of any flags change. Everything I have said was in the context of 'decimal' only implying format.

> Using 'decimal' to imply the real unit of the token makes the most sense to me.

IMHO this will not be naturally intuitive to users and will need to be very clearly documented.

> Right now we're implying that everything is the intified

Do you mean for just the `web3` command? Or throughout the whole api? The api might need updating to stay consistent.

Reply:
> We use gwei only in terms of gas prices and that's because that's what everyone uses

nit/ but gas prices can fluctuate and it's not unreasonable to imagine `Mwei (lovelace)` or  `microether (szabo)` being more convenient units, or even base token price inflating and  making `milliether (finney)` more convenient to send around.

Reply:
The web3 JS and RPC APIs actually both use `value` not `amount`, so we could switch to `-value` along with a new default unit so that it doesn't surprise anyone.

Reply:
I think value might be misleading since somebody in the know might expect value to be exactly what is passed along. Meaning our current "amount" would be more appropriately named "value". Trying to think of the best way to prevent mistakes.